### PR TITLE
Bindings for cha cha20 poly1305 variants

### PR DESCRIPTION
**Pull request by lmctv**
Only exposed as a low-level API, but should fulfill #308.

### CODE REVIEW

File: docs/vectors/index.rst [hunk 1]
```diff
@@ -30,6 +30,15 @@ The reference vector for testing the :py:class:`nacl.secret.SecretBox`
 implementation come from libsodium's ``test/default/secretbox.c``
 and the corresponding expected outputs in ``test/default/secretbox.exp``
 
+chacha20poly1305
+^^^^^^^^^^^^^^^^
+
+The reference vectors for both the legacy draft-agl-tls-chacha20poly1305
+and the IETF ratified rfc7539 chacha20poly1305 constructions are taken
+from libressl version 2.5.5 tests/aeadtests.txt, excluding the shortened
+authentication tag vectors, since libsodium does only support full sized
```
Comment (reaperhulk) [inline docs/vectors/index.rst]: s/does only support/only supports

Comment (lmctv) [reply docs/vectors/index.rst]: Corrected in the corresponding commit of the splitted-out PR #381

Comment (lmctv) [reply docs/vectors/index.rst]: Test parametrized on construction following your suggestion.

**Comment by reaperhulk** [comment]
This can be rebased now.

**Comment by lmctv** [comment]
Rebased, as requested.

**Comment by lmctv** [comment]
@reaperhulk something missing here?

**Comment by lmctv** [comment]
@reaperhulk I think the requested changes are both in. Something missing?

**Comment by maqp** [comment]
Thank you for this!

**Pull request merged**